core/userrights.class.inc.php

```diff
@@ -1921,50 +1921,51 @@ public static function FlushPrivileges($bResetAdminCache = false)
 	 */
 	protected static function FindUser($sLogin, $sAuthentication = 'any', $bAllowDisabledUsers = false)
 	{
-		if ($sAuthentication == 'any')
+		if ($sAuthentication === 'any')
 		{
 			$oUser = self::FindUser($sLogin, 'internal');
-			if ($oUser == null)
+			if ($oUser !== null)
 			{
-				$oUser = self::FindUser($sLogin, 'external');
+				return $oUser;
 			}
+
+			return self::FindUser($sLogin, 'external');
 		}
-		else
+
+		if (!isset(self::$m_aCacheUsers))
 		{
-			if (!isset(self::$m_aCacheUsers))
-			{
-				self::$m_aCacheUsers = array('internal' => array(), 'external' => array());
-			}
+			self::$m_aCacheUsers = [ 'internal' => array(), 'external' => array() ];
```

Comment:
Why not replace them all?
```suggestion
			self::$m_aCacheUsers = [ 'internal' => [], 'external' => [] ];
```